garrysmod/gamemodes/terrortown/gamemode/gamemsg.lua

```diff
@@ -189,6 +189,10 @@ function GM:PlayerCanHearPlayersVoice(listener, speaker)
    if listener:IsSpec() and listener.mute_team == speaker:Team() then
       return false, false
    end
+   
+   if listener:IsSpec() and listener.mute_team == 4 then

```

Comment:
This would be more clear if it was `listener.mute_team == MUTE_ALL`. You can move the MUTE_* constants defined in cl_voice.lua to shared.lua, so that they can be used here.

Same for the 0, 4, 5 used below.